python/rainbow/simulators/prox_soft_bodies/jacobi_parallel.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+import numba as nb
+from rainbow.util.timer import Timer
+
+
+@nb.njit(parallel=True, nogil=True, cache=True)
+def sweep(K, w, z, b, mu, x, friction_solver, hybrid = True):
+    """ Parallel Jacobi Proximal Algorithm.
+
+    Args:
+        K (int): Contact points number.
+        w (ArrayLike): The WJT.dot(x).
+        z (ArrayLike): The current contact force.
+        b (ArrayLike): b = Ju^t + ∆t JM^{-1}h + EJu^t
+        mu (float): The coefficient of friction.
+        x (ArrayLike): The current contact force.
+        friction_solver (callable): The friction cone function.
+        hybrid (bool): If True, use the hybrid parallel Jacobi algorithm.
+
+    Returns:
+        ArrayLike: The new contact force.
+    """
+    if not hybrid:
```

Comment:
Not sure I from this code alone get the point of the "hybrid" version of the sweep.